### PR TITLE
Password optional sign-in guide for .net

### PR DESCRIPTION
**Pull request by danielmaharry-okta**
## Description:
- **What's changed?** Password optional sign-in guide for dotnet
- **Is this PR related to a Monolith release?** No

Netlify (Sep 1, 1700 BST) : https://6310d84f73ca061508010da9--reverent-murdock-829d24.netlify.app/docs/guides/pwd-optional-sign-in-email/aspnet/main/

### Resolves:

* [OKTA-502077](https://oktainc.atlassian.net/browse/OKTA-502077)

### CODE REVIEW

**Review by susanharper-okta — APPROVED** [review]
Looks good - just have one comment about the diagram: 
packages/@okta/vuepress-site/.vuepress/public/img/pwd-optional/pwd-optional-dotnet-sign-in-summary.png

SERVER SIDE should be SERVER-SIDE

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsummary.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+<div class="full">
+
+![Sequence diagram illustrating the password-optional sign-in use case.](/img/pwd-optional/pwd-optional-dotnet-sign-in-summary.png)
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsummary.md:3]: Big empty space above the "Okta email challenge flow" box.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+```csharp
+var authnOptions = new AuthenticationOptions { Username = model.UserName };
+var authnResponse = await _idxClient
+   .AuthenticateAsync(authnOptions).ConfigureAwait(false);
+```
+
+### 3. The user verifies their identity with the email authenticator
+
+Query the `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `AuthenticateAsync()` to discover the current status of the authentication process. If you configured your Okta org correctly, the status is `AwaitingChallengeAuthenticatorSelection`, indicating that the user needs to verify their identity and must now choose the email authenticator to complete the identity challenge.
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md:25]: To bring the word count down somewhat, suggest rewording it with something like the following:

AuthenticateAsync() should return an AuthenticationResponse.AuthenticationStatus of AwaitingChallengeAuthenticatorSelection, which indicates the user needs to verify their identity and choose the email authenticator to complete the identity challenge.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+   default:
+      return View("Login", model);
+}
+```
+
+The email authenticator supports user verification by One-Time Password (OTP) and by magic links. To learn more, see the [Okta email integration guide](/docs/guides/authenticators-okta-email/aspnet/main/#integrate-email-challenge-with-magic-links).
+
+### 4. Your app handles an authentication success response
+
+When the user successfully completes the email authenticator challenge, the `AuthenticationStatus` property of the `AuthenticationResponse` object returned by `VerifyAuthenticatorAsync()` is `Success`. You can now call `AuthenticationHelper.GetIdentityFromTokenResponseAsync()` to retrieve the OIDC claims information about the user and pass them into your application. The user has now signed in.
```
Comment (robertnicolo-okta) [inline packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md:47]: Can we rewrite this sentence to be less wordy? I offered a suggestion in the other sign-up guide PR with a similar sentence.

**Review by robertnicolo-okta — CHANGES_REQUESTED** [review]
Looks good Dan. Just a handful of comments.

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsummary.md (hunk 1 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsummary.md:3]: fixed

File: packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md (hunk 2 above)
Comment (danielmaharry-okta) [reply packages/@okta/vuepress-site/docs/guides/pwd-optional-sign-in-email/main/aspnet/integrationsteps.md:25]: Changed to

`AuthenticateAsync()` returns an `AuthenticationResponse` object. Query its `AuthenticationStatus` property to discover the current status of the authentication process. A status of `AwaitingChallengeAuthenticatorSelection` indicates that the user needs to verify their identity with the email authenticator challenge.

**Review by robertnicolo-okta — APPROVED** [review]
lgtm
